src/interfaces/vault.ts

```diff
@@ -452,37 +466,29 @@ export class VaultInterface<T extends ChainId> extends ServiceInterface<T> {
     overrides: CallOverrides = {}
   ): Promise<TransactionResponse> {
     const signer = this.ctx.provider.write.getSigner(account);
-    const isZappingIntoPickleJar = PickleJars.includes(vault);
 
-    if (isZappingIntoPickleJar) {
+    if (this.isZappingIntoPickleJar({ vault })) {
       return this.zapIn(vault, token, amount, account, options, ZapProtocol.PICKLE, overrides);
     }
 
     const [vaultRef] = await this.getStatic([vault], overrides);
-    if (vaultRef.token === token) {
-      if (token === EthAddress) {
-        throw new SdkError("deposit:v2:eth not implemented");
-      } else {
-        const shouldUsePartner = this.shouldUsePartnerService(vault);
-        const vaultContract = new Contract(vault, VaultAbi, signer);
-
-        const makeTransaction = async (overrides: CallOverrides): Promise<TransactionResponse> => {
-          const tx = shouldUsePartner
-            ? await this.yearn.services.partner?.populateDepositTransaction(vault, amount, overrides)
-            : await vaultContract.populateTransaction.deposit(amount, overrides);
+    if (vaultRef.token !== token) {
+      return this.zapIn(vault, token, amount, account, options, ZapProtocol.YEARN, overrides);
+    }
 
-          if (!tx) {
-            throw new SdkError("deposit transaction failed");
-          }
+    if (token === EthAddress) {
+      throw new SdkError("deposit:v2:eth not implemented");
+    }
```

Comment:
We should remove this validation. Eth deposits are implemented, and are routed as zaps

Reply:
Good catch, this was there before but yeah, we are zapping to WETH. Removed with https://github.com/yearn/yearn-sdk/pull/280/commits/6ea9832bff17c9532c21ddab67a170fe01669a3c